src/zones.ts

```diff
@@ -68,25 +57,16 @@ export class Zones {
       this.unsubscribesForPaths[path]()
     }
     const tests = zones.map((zone) => {
-      const { upper, lower } = zone
-      if (upper !== undefined) {
-        if (lower !== undefined) {
-          return (value: number) => value < upper && value >= lower
-        } else {
-          return (value: number) => value < upper
-        }
-      } else {
-        if (lower !== undefined) {
-          return (value: number) => value > lower
-        } else {
-          return () => false
-        }
+      const { upper = Infinity, lower = -Infinity } = zone
+
+      return (value: Value) => {
+        return typeof value === 'number' && value < upper && value >= lower
```

Comment:
I had to add a `typeof value === 'number'` check to multiple of these tests so just refactored it quickly.  One thing I noticed is that one test had `value >= lower` and the other had `value > lower`. Was that intentional?